### PR TITLE
Add LoRA dropout and training mode support

### PR DESCRIPTION
**Pull request by aleroot**
## Proposed changes

Adding dropout support when training LoRA and DoRA adapters:

- Supports dropout values from native MLX and PEFT adapter configurations.
- Rejects invalid dropout values with a clear error.
- Preserves the correct quantization format when converting or combining adapters.
- Adds support for resuming training from saved adapter weights.
- Correctly switches between training and evaluation modes.
- Aligns DoRA calculations with the Python `mlx-lm` implementation.
- Keeps existing behavior unchanged when dropout is omitted or set to `0`.

#### Notes

Dropout temporarily hides a small, random part of the training input. This can help an adapter avoid memorizing its training data too closely. Dropout is automatically disabled when using the model for generation, so normal inference remains stable and repeatable.

## Checklist

- [x] I have read the [[CONTRIBUTING](https://github.com/ml-explore/mlx-swift-lm/blob/main/CONTRIBUTING.md)](https://github.com/ml-explore/mlx-swift-lm/blob/main/CONTRIBUTING.md) document
- [ ] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: Libraries/MLXLLM/LoraTrain.swift [hunk 1]
```diff
@@ -180,12 +185,19 @@ public enum LoRATrain {
...
+    ///
     /// ### See Also
     /// - ``loadLoRAData(directory:name:)``
     public static func evaluate(
         model: Module, dataset: [String], loss: LoraLossFunction = loss, tokenizer: Tokenizer,
         batchSize: Int, batchCount: Int
     ) -> Float {
+        let wasTraining = model.training
+        model.train(false)
+        defer { model.train(wasTraining) }
```
Comment (davidkoski) [inline Libraries/MLXLLM/LoraTrain.swift:199]: Nice find!  Just the other day I ran into this on https://github.com/ml-explore/mlx-swift/pull/418 -- I totally forgot about this mechanism!

File: Libraries/MLXLLM/LoraTrain.swift [hunk 2]
```diff
@@ -215,6 +227,21 @@ public enum LoRATrain {
...
+    /// layers before calling this method. Set ``Parameters/completedIterations`` when continuing
+    /// a run whose `iterations` value is the desired total number of iterations. This restores
+    /// adapter parameters only; callers that require exact optimizer continuation must persist and
+    /// restore optimizer state separately. The file must not contain keys that are absent from the
+    /// model's adapter structure.
+    public static func loadLoRAWeights(model: Module, url: URL) throws {
+        let weights = try MLX.loadArrays(url: url)
+        try model.update(
+            parameters: ModuleParameters.unflattened(weights),
+            verify: .noUnusedKeys)
```
Comment (davidkoski) [inline Libraries/MLXLLM/LoraTrain.swift:242]: This should probably eval the model after loading the weights (the load is lazy).

**Review by davidkoski — CHANGES_REQUESTED** [review]
Looks good, minor change suggested -- eval after load.

**Review by davidkoski — APPROVED** [review]
Looks good, thank you!

File: Libraries/MLXLLM/LoraTrain.swift (hunk 2 above)
Comment (aleroot) [reply Libraries/MLXLLM/LoraTrain.swift:242]: Thanks
